js/ui/shellDBus.js

```diff
@@ -516,3 +517,41 @@ var AppStoreService = class {
         this._dbusImpl.emit_signal('ApplicationsChanged', GLib.Variant.new('(as)', [allApps]));
     }
 };
+
+const HackableAppsManagerIface = loadInterfaceXML('com.endlessm.HackableAppsManager');
+
+var HackableAppsManager = class {
+    constructor() {
+        this._dbusImpl = Gio.DBusExportedObject.wrapJSObject(HackableAppsManagerIface, this);
+        const res = this._dbusImpl.export(Gio.DBus.session, '/com/endlessm/HackableAppsManager');
+        Gio.bus_own_name_on_connection(Gio.DBus.session, 'com.endlessm.HackableAppsManager',
+                                       Gio.BusNameOwnerFlags.REPLACE, null, null);
+
+        this._codeViewManager = Main.wm._codeViewManager;
+        this._codeViewManager.connect('session-state-changed',
+                                      Lang.bind(this, this._currentlyHackableAppsChanged));
+        this._codeViewManager.connect('session-app-lost',
+                                      Lang.bind(this, this._onSessionAppLost));
+    }
+
+    _currentlyHackableAppsChanged() {
+        const variant = new GLib.Variant('as', this.CurrentlyHackableApps);
+        this._dbusImpl.emit_property_changed('CurrentlyHackableApps', variant);
+    }
+
+    _onSessionAppLost(codeViewManager, session) {
+        if (session.state === Codeview.STATE_TOOLBOX)
+            this._currentlyHackableAppsChanged();
+    }
+
+    get CurrentlyHackableApps() {
+        let app_ids = [];
+        for (let session of this._codeViewManager._sessions) {
+            if (session.app && session.state === Codeview.STATE_TOOLBOX) {
```

Comment:
As mentioned elsewhere, we don't want to return only hackable apps showing a toolbox; we should include whether they're on the toolbox or app side as another piece of information through the API.